Subject: Linux-RT 2.6.25-rt
From: http://www.kernel.org/pub/linux/kernel/projects/rt/
Acked-by: Sven-Thorsten Dietrich <sdietrich@suse.de>
From mingo@elte.hu Mon Feb 11 19:43:38 2008
Date: Sun, 10 Feb 2008 08:20:13 +0100
From: Ingo Molnar <mingo@elte.hu>
To: linux-kernel@vger.kernel.org
Cc: Linus Torvalds <torvalds@linux-foundation.org>,
     Andrew Morton <akpm@zip.com.au>, Steven Rostedt <rostedt@goodmis.org>
Subject: [05/19] ftrace: make the task state char-string visible to all

From: Steven Rostedt <srostedt@redhat.com>

The tracer wants to be able to convert the state number
into a user visible character. This patch pulls that conversion
string out the scheduler into the header. This way if it were to
ever change, other parts of the kernel will know.

Signed-off-by: Steven Rostedt <srostedt@redhat.com>
Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 include/linux/sched.h |    2 ++
 kernel/sched.c        |    2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

Index: linux-2.6.24.4-rt4/include/linux/sched.h
===================================================================
--- linux-2.6.24.4-rt4.orig/include/linux/sched.h	2008-03-24 19:05:41.000000000 -0400
+++ linux-2.6.24.4-rt4/include/linux/sched.h	2008-03-24 19:05:41.000000000 -0400
@@ -2031,6 +2031,8 @@ static inline void migration_init(void)
 }
 #endif
 
+#define TASK_STATE_TO_CHAR_STR "RSDTtZX"
+
 #endif /* __KERNEL__ */
 
 #endif
Index: linux-2.6.24.4-rt4/kernel/sched.c
===================================================================
--- linux-2.6.24.4-rt4.orig/kernel/sched.c	2008-03-24 19:05:41.000000000 -0400
+++ linux-2.6.24.4-rt4/kernel/sched.c	2008-03-24 19:05:41.000000000 -0400
@@ -4847,7 +4847,7 @@ out_unlock:
 	return retval;
 }
 
-static const char stat_nam[] = "RSDTtZX";
+static const char stat_nam[] = TASK_STATE_TO_CHAR_STR;
 
 static void show_task(struct task_struct *p)
 {
